lib/src/ruby/api.cc

```diff
@@ -485,21 +486,47 @@ namespace facter { namespace ruby {
         }
         LOG_DEBUG("ruby was found at \"%1%\".", ruby);
 
-        bool success;
-        string output, none;
-        tie(success, output, none) = execute(ruby, { "-e", "print File.join(RbConfig::CONFIG['"+libruby_configdir()+"'], RbConfig::CONFIG['LIBRUBY_SO'])" });
+        bool success, loaded = false;
+        string ruby_cmd_args, ruby_output, libruby_file, libruby_full_path, none;
+        tie(success, libruby_file, none) = execute(ruby, { "-e", "print RbConfig::CONFIG['LIBRUBY_SO']"});
```

Comment:
How about:

```ruby
print(['libdir', 'archlibdir', "sitearchlibdir"].find do |name| dir = RbConfig::CONFIG[name]; next unless dir; file = File.join(dir, RbConfig::CONFIG['LIBRUBY_SO']); break file if File.exist? file end)
```

Then have the `['libdir', 'archlibdir', "sitearchlibdir"]` part inserted as a string based on `libruby_config_variables` (perhaps just returning this array as a string?)